x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/trial_license_complete_tier/execution_logic/machine_learning_alert_suppression.ts

```diff
@@ -0,0 +1,1082 @@
+              [ALERT_ORIGINAL_TIME]: firstTimestamp,
+              [ALERT_SUPPRESSION_DOCS_COUNT]: 0,
+            })
+          );
+
+          const secondTimestamp = new Date().toISOString();
+          const secondAnomaly = {
+            ...baseAnomaly,
+            timestamp: secondTimestamp,
+          };
+
+          // Add more anomalies, then disable and re-enable to trigger another
+          // rule run. The second anomaly should trigger an update to the
+          // existing alert without changing the timestamp
+          await indexListOfDocuments([secondAnomaly, secondAnomaly]);
+          await patchRule(supertest, log, { id: createdRule.id, enabled: false });
+          await patchRule(supertest, log, { id: createdRule.id, enabled: true });
+          const secondAlerts = await getOpenAlerts(
+            supertest,
+            log,
+            es,
+            createdRule,
+            RuleExecutionStatusEnum.succeeded,
+            undefined,
+            new Date()
+          );
+
+          expect(secondAlerts.hits.hits).toHaveLength(2);
+          expect(secondAlerts.hits.hits[1]._source).toEqual(
+            expect.objectContaining({
+              [ALERT_SUPPRESSION_TERMS]: [
+                {
+                  field: 'user.name',
+                  value: ['root'],
+                },
+              ],
+              [ALERT_ORIGINAL_TIME]: secondTimestamp,
+              [ALERT_SUPPRESSION_START]: secondTimestamp,
+              [ALERT_SUPPRESSION_END]: secondTimestamp,
+              [ALERT_SUPPRESSION_DOCS_COUNT]: 1, // 1 of the two new anomalies was suppressed on this execution
+            })
+          );
+        });
+      });
+
+      describe('with interval suppression duration', () => {
+        beforeEach(() => {
+          ruleProps = {
+            ...baseRuleProps,
+            alert_suppression: {
+              duration: {
+                value: 300,
+                unit: 'm',
+              },
+              group_by: ['user.name'],
+              missing_fields_strategy: 'suppress',
+            },
+          };
+        });
+
+        it('performs no suppression if a single alert is generated', async () => {
+          const timestamp = new Date().toISOString();
+          const anomaly = {
+            ...baseAnomaly,
+            timestamp,
+          };
+          await indexListOfDocuments([anomaly]);
+          const createdRule = await createRule(supertest, log, ruleProps);
+          const alerts = await getAlerts(supertest, log, es, createdRule);
+
+          expect(alerts.hits.hits).toHaveLength(1);
+          expect(alerts.hits.hits[0]._source).toEqual(
+            expect.objectContaining({
+              [ALERT_SUPPRESSION_TERMS]: [{ field: 'user.name', value: ['root'] }],
+              [ALERT_SUPPRESSION_DOCS_COUNT]: 0,
+            })
+          );
+        });
+
+        it('suppresses alerts across two executions', async () => {
```

Comment:
this test not only suppress on 2 rule executions but deduplicate new alert

Reply:
@vitaliidm this isn't an actionable comment. Is the described behavior expected? Do you want me to change the test, or simply update the test description?